app/pods/publications/publication/proofs/documents/controller.js

```diff
@@ -222,23 +235,30 @@ export default class PublicationsPublicationProofsDocumentsController extends Co
     await Promise.all(saves);
   }
 
-  async performSaveCorrectionDocument(proofDocument) {
+  async performSavePieceUpload(uploadProperties, isSource, isCorrection) {
     const now = new Date();
 
     const documentContainer = this.store.createRecord('document-container', {
       created: now,
     });
     await documentContainer.save();
 
-    const piece = this.store.createRecord('piece', {
+    const pieceProperties = {
       created: now,
       modified: now,
-      file: proofDocument.file,
+      name: uploadProperties.name,
       confidential: false,
-      name: proofDocument.name,
+      file: uploadProperties.file,
       documentContainer: documentContainer,
-      publicationSubcaseCorrectionFor: this.publicationSubcase,
-    });
+    };
+    if (isSource) {
+      pieceProperties.publicationSubcaseSourceFor = this.publicationSubcase;
+    } else if (isCorrection) {
+      pieceProperties.publicationSubcaseCorrectionFor = this.publicationSubcase;
+    } else {
+      throw new Error('no relationship specified');
```

Comment:
Deze error, die ga je normaal nooit hitten?
Er is ook geen logica dat deze error gaat opvangen/afhandelen dus je data (file en document-container) is niet verwijderd in dit geval.
Ik zou dit stuk eerder aanpassen naar:

```
    if (isCorrection) {
      pieceProperties.publicationSubcaseCorrectionFor = this.publicationSubcase;
    } else
      pieceProperties.publicationSubcaseSourceFor = this.publicationSubcase;
    }
```
Zo hoeven we maar 1 boolean te checken, en vallen we sowieso terug op een default moest er toch iets mis zijn.